BACKPORT - change signature of ipath_get_sb for 2.6.17 and earlier

Signed-off-by: Ralph Campbell <ralph.campbell@qlogic.com>

---
 drivers/infiniband/hw/ipath/ipath_fs.c |   15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

Index: ofed_kernel/drivers/infiniband/hw/ipath/ipath_fs.c
===================================================================
--- ofed_kernel.orig/drivers/infiniband/hw/ipath/ipath_fs.c
+++ ofed_kernel/drivers/infiniband/hw/ipath/ipath_fs.c
@@ -362,14 +362,13 @@ bail:
 	return ret;
 }
 
-static int ipathfs_get_sb(struct file_system_type *fs_type, int flags,
-			const char *dev_name, void *data, struct vfsmount *mnt)
-{
-	int ret = get_sb_single(fs_type, flags, data,
-				    ipathfs_fill_super, mnt);
-	if (ret >= 0)
-		ipath_super = mnt->mnt_sb;
-	return ret;
+static struct super_block *ipathfs_get_sb(struct file_system_type *fs_type,
+					  int flags, const char *dev_name,
+					  void *data)
+{
+	ipath_super = get_sb_single(fs_type, flags, data,
+				    ipathfs_fill_super);
+	return ipath_super;
 }
 
 static void ipathfs_kill_super(struct super_block *s)
